### PR TITLE
feat: test runPartial deepEquals run

### PR DESCRIPTION
**Pull request by WilcoFiers**

### CODE REVIEW

**Comment by AdnoC** [comment]
Won't this test have to be changed every time we change axe-core (e.g. add a rule or check)? Do we want to add that to the integration update process?

**Comment by nloding** [comment]
This test may also fail in other integrations, depending on what frames are available to inject axe into. I think on the C# side there are some issues with selecting shadow DOM frames from the webdriver. I think that means it's possible for `getFrameContexts` and `runPartial` to generate data from a frame not accessible by the webdriver.

**Comment by AdnoC** [comment]
This test needs to be worked out more (and tested on real 4.0.3 vs 4.3.x results objects). There are many differences that will need to be normalized that have nothing to do with the scan results and just due to the version of axe-core. The easiest to normalize is `results.testEngine.version`. Harder is *every single help URL*, which has the version of axe-core as part of the URL path. There may be more things that I haven't thought of.

**Comment by WilcoFiers** [comment]
@AdnoC This is comparing the results of the same axe-core version. Deleting runPartial and finishRun should force the integration into using `axe.run()`, but the results, except for the timestamp should be identical. If it isn't, something's up.

@nloding Do we have integrations that can't test recursive iframes? I know Cypress doesn't, other than that I didn't think so. I picked this particular test file because I thought this was one that would work everywhere. If not I think it's okay to change which file is tested, as long as we add a comment to the test explaining why.

**Comment by AdnoC** [comment]
Ah. You are correct. In that case I approve of the test.

File: README.md [hunk 1]
```diff
@@ -137,6 +137,18 @@ describe('frame tests', () => {
       'input'
     ]);
   });
+
+  it('returns the same results from runPartial as from legacy mode', async () => {
+    await driver.get(`${addr}/nested-iframes.html`);
+    const normalResults = await new AxeBuilder(driver, axeSource).analyze();
+
+    const legacyResults = await new AxeBuilder(driver, axeSource + `;
+      delete window.axe.runPartial;
```
Comment (straker) [inline README.md]: I'd like to make the test a bit more robust and ensure that the `delete` happens as we expect. If for some reason the `delete` fails, we'd never know it ran `runPartial` twice. However `throwing` at this point doesn't stop the run from happening so I'm not sure what we can do.

**Review by AdnoC — APPROVED** [review]

**Pull request merged**